### ISSUE TITLE
BUG: Skyvern sensitive to port change

### ISSUE THREAD

**Issue by gwpl** [Skyvern-AI/skyvern#1277]
When I changed ports in docker-compose file to others , then Skyvern UI create new task started returning "network error" box .

However, after reverting this change and restarting, skyvern was working back again on default ports.

**Comment by prajwal2431** [comment]
is this issue still open i want to work on it

**Comment by maltorfer** [comment]
Yes. I have a setup a reverse proxy to access the UI. When I check the chrome developer, I see lots of connections to "localhost:8000" which obviously doesn't work. Perhaps you make a setting where you can configure the base URL for the API.

**Comment by gwpl** [comment]
still not working, example port change:

```
diff --git a/docker-compose.yml b/docker-compose.yml
index a94b875..8e0ce9a 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -3,8 +3,8 @@ services:
     image: postgres:14-alpine
     restart: always
     # comment out if you want to externally connect DB
-    # ports:
-    #   - 5432:5432
+    ports:
+      - 5048:5432
     volumes:
       - ./postgres-data:/var/lib/postgresql/data
     environment:
@@ -23,7 +23,7 @@ services:
     restart: on-failure
     # comment out if you want to externally call skyvern API
     ports:
-      - 8000:8000
+      - 5049:8000
     volumes:
       - ./artifacts:/data/artifacts
       - ./videos:/data/videos
@@ -59,8 +59,8 @@ services:
     image: public.ecr.aws/skyvern/skyvern-ui:latest
     restart: on-failure
     ports:
-      - 8080:8080
-      - 9090:9090
+      - 5050:8080
+      - 5051:9090
     volumes:
       - ./artifacts:/data/artifacts
       - ./videos:/data/videos
```

**Comment by prajwal2431** [comment]
To resolve the issue caused by changing ports in the `docker-compose.yml` file, ensure that all related environment variables, such as `VITE_WSS_BASE_URL`, are updated to reflect the new ports. For example, if the Skyvern API port changes to `5049`, set `VITE_WSS_BASE_URL=ws://localhost:5049/api/v1`. After updating the configuration, rebuild and restart the containers using `docker-compose down` and `docker-compose up --build`. Clear your browser cache or perform a hard refresh (`Ctrl+Shift+R`) to load the updated settings. Additionally, verify service communication by checking the logs (`docker-compose logs -f`). Ensuring consistency across port configurations should resolve the "network error" issue.

**Comment by orapaorapa** [comment]
[docker-compose(dot)yml.txt](https://github.com/user-attachments/files/18186341/docker-compose.dot.yml.txt)

After rebuilding with 'docker-compose up -build' , i launch the navigation panel on 192.168.1.32:8080 , works fine ...but when i try 'new task' i have : unexpected application error! 404 not found .  The second error is when i try from prompt : 'error creating task from prompt : request failed with status error 400' . ??? thx

**Comment by orapaorapa** [comment]
[docker-compose(dot)yml.txt](https://github.com/user-attachments/files/18186426/docker-compose.dot.yml.txt)
After rebuilding with 'docker-compose up -build' , i launch the navigation panel on 192.168.1.32:8080 , works fine ...but when i try 'new task' i have : unexpected application error! 404 not found . The second error is when i try from prompt : 'error creating task from prompt : request failed with status error 400' . ??? thx

**Comment by rublev** [comment]
> [docker-compose(dot)yml.txt](https://github.com/user-attachments/files/18186426/docker-compose.dot.yml.txt) After rebuilding with 'docker-compose up -build' , i launch the navigation panel on 192.168.1.32:8080 , works fine ...but when i try 'new task' i have : unexpected application error! 404 not found . The second error is when i try from prompt : 'error creating task from prompt : request failed with status error 400' . ??? thx

Same. Bare 404's. Nothing works! If running without docker, you get a whole different set of errors with the "secondary LLM" or w.e

**Comment by deveshruttala-edu** [comment]
👋 Hey there! This looks like something Hyperbrowser could assist with. We're building an AI-native browser that automates workflows, integrates smart tools, and enhances productivity directly inside the browser. If you're tackling tasks like this often, Hyperbrowser might be just what you need.

🚀 Learn more and try it out: https://www.hyperbrowser.ai/
We’d love your feedback!
